### ISSUE TITLE
[Feature]: Implement binlog compression on minio

### ISSUE THREAD

**Issue by xiaofan-luan** [milvus-io/milvus#15561]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your feature request related to a problem? Please describe.

We've see many people complain about Milvus 2.0 spent too many disk spaces.

One thing we can do is do compress the binlogs and check if we can get some compression ratio, If so compression should also help the load speed due to saved IOs

**Comment by congqixia** [comment]
One good document for binlog format is [this doc](https://github.com/milvus-io/milvus/blob/master/docs/developer_guides/chap08_binlog.md)
Since there is some meta information in the binlog header, I would suggest to keep the binlog headers untouched and compress the data sections. Also the compression method shall be an  extra field in the future binlog header as well.

**Comment by xiaofan-luan** [comment]
> One good document for binlog format is [this doc](https://github.com/milvus-io/milvus/blob/master/docs/developer_guides/chap08_binlog.md) Since there is some meta information in the binlog header, I would suggest to keep the binlog headers untouched and compress the data sections. Also the compression method shall be an extra field in the future binlog header as well.

Agree.  I would suggest to enable compression on collection level. And each binlog header is not compressed. When read bin log, one need to parse binlog header and check if it's compressed or not.

**Comment by xiaofan-luan** [comment]
@yah01 zstd level 3 should be well enough for the recommended settings.
